packages/medusa/src/api/routes/admin/stock-locations/get-stock-location.ts

```diff
@@ -73,7 +76,9 @@ export default async (req: Request, res: Response) => {
     )
   }
 
-  let stockLocation = await locationService.retrieve(id, retrieveConfig)
+  let stockLocation = await locationService.retrieve(id, retrieveConfig, {
+    transactionManager: manager,
+  })
 
   if (includeSalesChannels) {
     const [location] = await joinSalesChannels(
```

Comment:
todo: we are missing the manager here. Under the hood it calls the channelLocationService and the stockLocationService without manager. Therefore another transaction will be created